### PR TITLE
21824 Code changes to support continuation application resubmission

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#21824

*Description of changes:*
- app version = 5.11.9
- added vue-auto-resize package
- allow filings in Change Requested status for temp id case only
- store filing status (tombstone, getter, action)
- added font size base style
- added Change Requested message box
- renamed isShowReviewConfirmBtn() -> isShowNextBtn()
- updated File and Pay button label
- updated Filing Status enum
- updated misc interfaces for new header property (latestReviewComment)
- now restore expro registration confirmation checkbox (isConfirmed)
- waive fees when filing is Change Requested
- updated isContinuationValid()
- only allow change to effective date/time when not Draft
- only show staff payment step when filing is Draft
- misc cleanup
- added/updated unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-722-p6p43anx.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-722-p6p43anx.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-722-p6p43anx.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-722-p6p43anx.web.app

File: src/views/ContinuationIn/ContinuationInReviewConfirm.vue [hunk 1]
```diff
@@ -96,6 +97,18 @@
...
+      id="continuation-effective-date-time-section"
+      class="mt-10"
+    >
+      <header>
+        <h2>Continuation Effective Date and Time</h2>
+        <p class="mt-4">
+          The incorporation date and time will be upon approval.
+        </p>
+      </header>
+    </section>
```
Comment (severinbeauvais) [inline src/views/ContinuationIn/ContinuationInReviewConfirm.vue:111]: New section to show if filing status isn't draft or isn't future effective. In other words, if the filing is Change Requested or Immediate.

File: src/views/ContinuationIn/ContinuationInReviewConfirm.vue [hunk 2]
```diff
@@ -183,7 +196,7 @@
 
     <!-- Staff Payment -->
     <section
-      v-if="isRoleStaff"
+      v-if="isRoleStaff && getFilingStatus === FilingStatus.DRAFT"
```
Comment (severinbeauvais) [inline src/views/ContinuationIn/ContinuationInReviewConfirm.vue:199]: Don't show staff payment section if the filing is Change Requested. You'll see in filing-template-mixin that we hard-code the filing data to "waive fees" if we're resuming a Change Requested filing. This sets the Fee Summary to $0, ie:

image

File: src/mixins/filing-template-mixin.ts [hunk 3]
```diff
@@ -558,14 +561,17 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
       if (effectiveDate >= this.getCurrentJsDate) this.setEffectiveDate(effectiveDate)
     }
 
-    // NB: Staff role is mutually exclusive with premium account.
-    if (this.isRoleStaff) {
+    if (status === FilingStatus.CHANGE_REQUESTED) {
+      // if user is editing a (previously filed) filing, waive fees as they don't have to pay again
+      this.parseStaffPayment({ header: { waiveFees: true } } as any)
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:566]: This is to zero out the Fee Summary. The actual determination of whether to redirect to Pay or not is in Actions.vue, and it's actually the Legal API that tells us if we have to pay or skip pay.

**Review by ArwenQin — APPROVED** [review]
Looks good!

**Review by ketaki-deodhar — APPROVED** [review]
Looking good!

**Pull request merged**